cmd/nmi/main.go

```diff
@@ -133,14 +132,13 @@ func main() {
 		redirector = server.LinuxRedirector(s)
 	}
 
-	go redirector(s, subRoutinedone, mainRoutineDone)
+	go redirector(s, mainRoutineDone)
 
 	if err := s.Run(); err != nil {
 		klog.Fatalf("%s", err)
 	}
 
 	close(mainRoutineDone)
```

Comment:
You basically reintroduced the issue that main function will not wait for the Sync() exit.
Please bring the channel back. You need to think about how to keep the ownership of the channel in redirector code and close the channel instead of sending single message.

Higher level issue is that top level functions do not have any unit and integration testing to catch issues like this. So unless someone mentions this in a code review it will go unnoticed

Reply:
That is a bad push by accident... Reverted the change and add the direction for the channel.